generate.ts

```diff
@@ -187,24 +189,36 @@ function extractArgs(args, argsByName, targetOutputPathList, errorMessages) {
     }
 
     // Pull from environment variables if there's no console-defined targets
-    if (targetOutputPathList.length === 0 && process.env.hasOwnProperty("SdkSource") && process.env.hasOwnProperty("SdkName")) {
-        checkTarget(process.env.hasOwnProperty("SdkSource"), process.env.hasOwnProperty("SdkName"), targetOutputPathList, errorMessages);
+    if (!buildTarget.destPath) {
+
+        console.log("argsByName: " + JSON.stringify(argsByName) + " " + argsByName["destpath"]);
+
+        if (argsByName["destpath"]) {
+            checkTarget(argsByName["srcfolder"], argsByName["destpath"], buildTarget, errorMessages);
+        } else if (process.env.hasOwnProperty("SdkName")) {
+            checkTarget(process.env["SdkSource"], process.env["SdkName"], buildTarget, errorMessages);
+        }
     }
 }
 
-interface ITargetOutput {
-    name: string,
-    dest: string,
+interface IBuildTarget {
+    buildFlags: string[],
+    destPath: string,
+    srcFolder: string,
+    versionKey: string,
+    versionString: string,
 }
-function checkTarget(sdkSource, sdkDestination, targetOutputPathList, errorMessages) {
-    var targetOutput: ITargetOutput = {
-        name: sdkSource,
-        dest: path.normalize(sdkDestination)
-    };
-    if (fs.existsSync(targetOutput.dest) && !fs.lstatSync(targetOutput.dest).isDirectory())
-        errorMessages.push("Invalid target output path: " + targetOutput.dest);
-    else
-        targetOutputPathList.push(targetOutput);
+function checkTarget(sdkSrcFolder, sdkDestination, buildTarget: IBuildTarget, errorMessages) {
```

Comment:
>checkTarget [](start = 9, length = 11)

This sounds more like "setBuildTarget" rather than "checkTarget". Well, we are setting it there, not just checking something.